src/lib/QueryType/ContentQueryType.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Personalization\QueryType;
+
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\QueryType\OptionsResolverBasedQueryType;
+use Ibexa\Personalization\Config\Repository\RepositoryConfigResolverInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class ContentQueryType extends OptionsResolverBasedQueryType
+{
+    /** @var \Ibexa\Personalization\Config\Repository\RepositoryConfigResolverInterface */
+    private $repositoryConfigResolver;
+
+    public function __construct(RepositoryConfigResolverInterface $repositoryConfigResolver)
+    {
+        $this->repositoryConfigResolver = $repositoryConfigResolver;
+    }
+
+    public static function getName(): string
+    {
+        return 'Ibexa:Personalization:ContentQueryType';
+    }
+
+    protected function configureOptions(OptionsResolver $optionsResolver): void
+    {
+        $optionsResolver
+            ->setDefaults(
+                [
+                    'contentId' => null,
+                    'language' => null,
+                ]
+            )
+            ->setAllowedTypes('contentId', ['int', 'string'])
+            ->setAllowedTypes('language', ['null', 'string']);
+    }
+
+    /**
+     * @phpstan-param array{
+     *  'contentId': int|string,
```

Comment:
Content item ID is always `int`
```suggestion
     *  'contentId': int,
```